WorkoutGeneratorApp/models/ExerciseModel.ts

```diff
@@ -0,0 +1,68 @@
+/* eslint-disable prettier/prettier */
+/* One set of an exercise, denoted by W x R. */
+interface ExerciseSet {
+  readonly Weight: number;
+  readonly Reps  : number;
+  IsCompleted    : boolean;
+  Type           : ExerciseType;
+  MuscleFocus    : MuscleGroup;
+  IntensityLevel : ExerciseIntensity;
+}
+
+/* Base Exercise model. */
+export interface Exercise {
+  readonly Name  : string;
+  readonly Sets  : ReadonlyArray<Omit<ExerciseSet, 'IsCompleted'>>;
+  readonly Unit  : MeasurementSystem;
+  readonly Image?: unknown; //TBD
+  readonly Media?: unknown; //TBD
+  RestTime?      : number;  //TBD, editable by user for long rest periods?
+}
+
+/* ExerciseGenerate */
+export interface ExerciseGenerate extends Exercise {
+  IsLocked: boolean;
+}
+
+/* ExerciseDo */
+export interface ExerciseDo extends Omit<Exercise, 'Sets'> {
+  Sets: Array<ExerciseSet>;
+}
+
+export enum ExerciseType {
+  All          = 'All',
+  Barbell      = 'Barbell',
+  Dumbbell     = 'Dumbbell',
+  Kettlebell   = 'Kettlebell',
+  Machine      = 'Machine',
+  Cable        = 'Cable',
+  MedicineBall = 'Medicine Ball',
+  Bands        = 'Bands',
+  Bodyweight   = 'Bodyweight',
+  Aerobic      = 'Aerobic',
+}
```

Comment:
I think we can add a 'core' section here as well!

Reply:
Yes we can!